libraries/plugins/query_txid/query_txid_plugin.cpp

```diff
@@ -0,0 +1,210 @@
+#include <graphene/query_txid/query_txid_plugin.hpp>
+#include <fc/io/fstream.hpp>
+#include <graphene/chain/transaction_entry_object.hpp>
+#include <queue>
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <leveldb/db.h>
+#include <leveldb/write_batch.h>
+#include <fc/signals.hpp>
+
+namespace graphene{ namespace query_txid{
+
+namespace detail
+{
+using namespace leveldb;
+class query_txid_plugin_impl
+{
+  public:
+    query_txid_plugin_impl(query_txid_plugin &_plugin)
+        : _self(_plugin)
+    {
+    }
+    ~query_txid_plugin_impl()
+    {
+    }
+
+    void collect_txid_index(const signed_block &b);
+
+    graphene::chain::database &database()
+    {
+        return _self.database();
+    }
+    void init();
+    static optional<trx_entry_object> query_trx_by_id(std::string txid);
+    std::string db_path = "trx_entry.db";
+    uint64_t limit_batch = 1000; //limit of leveldb batch
+
+  private:
+    query_txid_plugin &_self;
+
+    fc::signal<void()> sig_db_write;
+    fc::signal<void(const uint64_t)> sig_remove;
+
+    static leveldb::DB *leveldb;
+    void consume_block();                               //Consume block
```

Comment:
comment dont add additional value than the function name, please remove, check the others in this file.